mobius/examples/crinkle_arc.rs

```diff
@@ -1,99 +1,109 @@
+        GeneralizedCircle::Line(_) => panic!("Not implemented: sub arc that's a line"),
     };
 
-    let orthog_center = Complex::from_polar(p, angle_bisector);
+    // My convention is to compute the sub arc that's sweeping in the same
+    // angular direction as the original arc. But if the original one went from b -> a,
+    // now we're going from a -> b;
+    let angle_a_raw = orthog_circle.get_angle(a).unwrap();
+    let angle_b_raw = orthog_circle.get_angle(b).unwrap();
+    let sub_angles = ArcAngles::from_raw_angles(angle_b_raw, angle_a_raw, arc.direction());
 
-    Circle {
-        center: orthog_center,
-        radius: orthog_radius,
-    }
+    CircularArc::new(orthog_circle, sub_angles)
 }
 
-fn arc_lerp(circle: Circle, a: Complex, b: Complex, t: f64) -> Complex {
-    let angle_a = circle.get_angle(a).unwrap();
-    let angle_b = circle.get_angle(b).unwrap();
-
-    let lerp_angle = (1.0 - t) * angle_a + t * angle_b;
-
-    circle.get_point(lerp_angle)
+struct ArcFractal {
+    /// The original arc, a -> c
+    arc: CircularArc,
+    /// Two transformations (translate/rotate/scale) that shrink arc onto
+    /// the two respective sub-arcs
+    xforms: (Mobius, Mobius),
+    /// An arc orthogonal to the first one, as I want to render this too.
+    orthog_arc: CircularArc,
+    /// The two sub-arcs. The first one is c -> b, the second is b -> a
+    sub_arcs: (CircularArc, CircularArc),
 }
 
-fn arc_fractal(arc: CircularArc) -> (Mobius, Mobius) {
-    let CircularArc { circle, angles } = arc;
-    let ArcAngles(angle_a, angle_c) = angles;
-    let t = 0.5;
+impl ArcFractal {
+    pub fn new(arc: CircularArc, t: f64) -> Self {
+        let a = arc.start();
+        let b = arc.interpolate(t);
+        let c = arc.end();
 
-    let a = circle.get_point(angle_a);
-    let angle_b = t * (angle_a + angle_c);
-    let b = circle.get_point(angle_b);
-    let c = circle.get_point(angle_c);
+        let arc_cb = compute_orthogonal_arc(arc, b, c);
+        let arc_ba = compute_orthogonal_arc(arc, a, b);
 
-    let circle_ab = get_orthog_circle(circle, a, b);
-    let circle_bc = get_orthog_circle(circle, b, c);
+        let d = arc_ba.interpolate(t);
+        let e = arc_cb.interpolate(t);
 
-    let d = arc_lerp(circle_ab, b, a, t);
-    let e = arc_lerp(circle_bc, c, b, t);
+        let xform_bda = map_triple((a, b, c), (b, d, a)).unwrap();
+        let xform_ceb = map_triple((a, b, c), (c, e, b)).unwrap();
 
-    let xform_bda = map_triple((a, b, c), (b, d, a)).unwrap();
-    let xform_ceb = map_triple((a, b, c), (c, e, b)).unwrap();
-
-    (xform_bda, xform_ceb)
+        Self {
+            xforms: (xform_ceb, xform_bda),
+            arc,
+            orthog_arc: compute_orthogonal_arc(arc, a, c),
+            sub_arcs: (arc_cb, arc_ba),
+        }
+    }
 }
 
 fn main() -> Result<(), Error> {
-    let angles = ArcAngles::new(0.0, PI / 2.0).unwrap();
+    let angles = ArcAngles::new(-PI / 2.0, PI / 4.0).unwrap();
```

Comment:
This didn't work when I used an angle greater than 180 degrees. Investigate.

Reply:
It's because I'm assuming the inside arc is in the same direction as the original arc. That's not true if the original arc is > pi.

`ArcAngles` should have a `get_central_angle` which computes `|b - a|`